workflows/differentialabundance.nf

```diff
@@ -454,6 +478,35 @@ workflow DIFFERENTIALABUNDANCE {
             .mix(GSEA_GSEA.out.versions)
     }
 
+    if (params.gprofiler2_run) {
+
+        // For gprofiler2, use only features that are considered differential
+        ch_filtered_diff = FILTER_DIFFTABLE.out.filtered
+
+        if (!params.gprofiler2_background_file) {
+            // If deactivated, use empty list as "background"
+            ch_background = []
+        } else if (params.gprofiler2_background_file == "auto") {
+            // If auto, use input matrix as background
+            ch_background = CUSTOM_MATRIXFILTER.out.filtered.map{it.tail()}.first()
+        } else {
+            ch_background = Channel.from(file(params.gprofiler2_background_file, checkIfExists: true))
+        }
+
+        if (params.gprofiler2_token || params.gprofiler2_organism) {
+            // For gprofiler2, token and organism have priority and will override a gene_sets file
+            ch_gprofiler_gene_sets = []
+        } else {
+            ch_gprofiler_gene_sets = ch_gene_sets.toList()
+                .map{it -> it.size() > 1 ? error("gprofiler2 currently only accepts a single gene sets file!") : it.first()}            
+        }
+
+        GPROFILER2_GOST(
+            ch_filtered_diff,
+            ch_gprofiler_gene_sets,
+            ch_background
+        )
```

Comment:
```suggestion
        // For gprofiler2, token and organism have priority and will override a gene_sets file

        GPROFILER2_GOST(
            ch_filtered_diff,
            ch_gene_sets,
            ch_background
        )
```

I've often had trouble conditioning on channel sizes at runtime- I don't think it works (or at least not well). But we can do the check outside of the workflow logic along with the other input checks- see the other suggestion.

But the beauty of the priorities in the module is that you can just pass the gene sets straight through, and they'll be ignored if the other options are activated- so this gets a lot simpler.